lib/processors/grouping.ts

```diff
@@ -14,68 +20,91 @@ export async function addGrouping(
   }
 
   console.log("Adding grouping to deprecations...");
-  let { groups } = config;
+  let { groups } = config as { groups: Group[] };
 
   let deprecationsWithGroup: Deprecation[] = [];
 
   for (const deprecation of rawDeprecations) {
     const deprecationHasExistingGroup = groups.find((group) => {
       // If matchers are present test them else return false
       return group.matchers.length ?
-        group.matchers.some(
-        (reg) => reg && new RegExp(reg).test(deprecation.deprecationMessage)
-      ) : false;
+        group.matchers.some((reg) => testMessage(reg, deprecation.deprecationMessage)) : false;
     })?.key;
 
     if (deprecationHasExistingGroup) {
       deprecationsWithGroup.push({
         ...deprecation,
-        group: deprecationHasExistingGroup,
+        group: deprecationHasExistingGroup
       });
       continue;
     }
 
     const answer: { key: string; regexp: string } = await prompt([
       {
         // TODO: use autocomplete here? https://github.com/enquirer/enquirer/tree/master/examples/autocomplete
-        // Problem: can't have a custom input that is not in choices
+        // @Notice Problem: can't have a custom input that is not in choices
         type: "input",
         name: "key",
         message:
-          `Add group to deprecation ${deprecation.path}#${deprecation.lineNumber}` +
+          `Add group name to deprecation ${deprecation.path}#${deprecation.lineNumber}` +
           EOL +
           deprecation.deprecationMessage +
+          EOL +
+          `Think of the headline of the grouping file. The string will be kebab cased and used as file name for the group.` +
+          EOL +
+          `An example for a name could be 'Internal implementation detail' the filename will be 'internal-implementation-detail.md` +
```

Comment:
> An example for a name could be 'Internal implementation detail' the filename will be 'internal-implementation-detail.md` 

Currently this is not the case because all deprecations will be saved to one file.
We the option to change the text, or to change the implementation.

Do you think the markdown adds value? If not, we can remove it and it isn't a problem anymore 😁

Reply:
:)

Reply:
@timdeschryver 
I updated the text. Let me know what else is missing

Reply:
I think it can be merged :)